### ISSUE TITLE
[Bug]: Registered scheme doesn't honor privileges

### ISSUE THREAD

**Issue by aadcg** [electron/electron#41032]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

28.0.0

### What operating system are you using?

Other Linux

### Operating System Version

nix-channel (Nix) 2.16.1

### What arch are you using?

x64

### Expected Behavior

One of the sound track wouldn't play.

### Actual Behavior

Both sound tracks play.

### Testcase Gist URL

https://gist.github.com/aadcg/9e78ee363ec9046128b6b01344c95576

**Comment by codebytere** [comment]
@aadcg does this also happen on any other platforms? Or exclusively linux?

**Comment by codebytere** [comment]
@aadcg oh actually glad you clarified - you're right. I can in fact reproduce on macOS - i'd misread and thought that the expected behavior was both tracks playing. I'll update the issue and try to take a deeper look!

**Comment by codebytere** [comment]
@aadcg ok - had a deep dive and I think the issue is in large part that the function of the `stream` option has not been effectively documented and maintained over the years. Initially, it was added to fix a bug wherein audio and video tags did not play at all - the stream flag told the video/audio player to use stream behaviors instead of fully-buffered behaviors.

As of https://github.com/electron/electron/pull/25324 - it seems that in the scenario you've provided the audio will always play if it's fully buffered, despite not using fully-streaming behavior. At this point, we might want to consider removing it entirely as it's likely more misleading than useful. 

@deepak1556 what are your thoughts here?

**Comment by codebytere** [comment]
@aadcg yes - fair enough 😁 I do think it's also likely related to usage of `protocol.handle` and `net.fetch`, which didn't exist prior to v24. I'm still sussing that out.

> I've been running this gist since ~ July 2023 and it used to work great 

I tried to reproduce on macOS and both tracks play for me on all 25 versions and newer - was this not the case for you on Linux? Or specifically - I wasn't able to find a version using your repro that produces the expected behavior. Am I maybe missing something? What version were you running your repro on that it worked as expected?

**Comment by jmercouris** [comment]
I concur, I saw it functioning.
